internal/renderers/skia/lib.rs

```diff
@@ -60,42 +64,48 @@ pub struct SkiaRenderer {
     path_cache: ItemCache<Option<(Vector2D<f32, PhysicalPx>, skia_safe::Path)>>,
     rendering_metrics_collector: RefCell<Option<Rc<RenderingMetricsCollector>>>,
     rendering_first_time: Cell<bool>,
-    surface: DefaultSurface,
+    surface: once_cell::unsync::OnceCell<Box<dyn Surface>>,
+    window_handle_provider: Box<dyn raw_window_handle::HasWindowHandle>,
+    display_handle_provider: Box<dyn raw_window_handle::HasDisplayHandle>,
 }
 
 impl SkiaRenderer {
     /// Creates a new renderer is associated with the provided window adapter.
     pub fn new(
-        window_handle: raw_window_handle::WindowHandle<'_>,
-        display_handle: raw_window_handle::DisplayHandle<'_>,
-        size: PhysicalWindowSize,
+        window_handle_provider: impl raw_window_handle::HasWindowHandle + 'static,
+        display_handle_provider: impl raw_window_handle::HasDisplayHandle + 'static,
```

Comment:
I'm not entirely sure about the API here. In practice we end up boxing an Rc here. It would be nice to take an `Rc<dyn ...>`, but that requires some more gymnastics in the winit backend.